app/src/main/res/layout/activity_input_interaction_view_test.xml

```diff
@@ -35,15 +35,27 @@
       android:layout_height="wrap_content"
       android:layout_margin="8dp"
       android:background="@drawable/edit_text_background"
-      android:digits="0123456789."
       android:focusable="true"
       android:hint="Write the digit here."
-      android:inputType="numberDecimal"
       android:longClickable="false"
       android:maxLength="200"
       android:padding="8dp"
       android:singleLine="true"
-      android:text="@={numericInputViewModel.answerText}" />
+      android:text="@={numericInputViewModel.answerText}"
```

Comment:
`android:inputType="numberSigned|numberDecimal"` 

This attribute helps to make sure that the number keyboard is shown in which you can use following characters, `-`, `.` and `0123456789` Also, it handles a lot of things on it own. Try using this, it might help more.

Reply:
This was suggested by ben in fraction input to keep it open for testing purpose.